### PR TITLE
mixed precision exchange

### PR DESCRIPTION
**Pull request by mcallic2**
**Description**
This pr includes changes to add mixed precision capabilities, all real values and math will stay in 64bit precision
This also changes variable declarations to be more aligned

Fixes # (issue)

**How Has This Been Tested?**
autotools with gcc/13.1.0, netcdf/4.9.2, mpich/4.1.2, hdf5/1.14.1-2, libyaml/0.2.5
 and with oneapi/2023.2, compiler-rt/2023.2.1, compiler/2023.2.1, hdf5/1.14.1-2, libyaml/0.2.5,
        tbb/2021.10.0, oclfpga/2023.2.1, mpi/2021.10.0, netcdf/4.9.2

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by mlee03** [comment]
@mcallic2 could you merge in mixedmode?

**Comment by mcallic2** [comment]
@mlee03 I merged in mixedmode, fingers crossed I did it correctly lol

File: .github/workflows/build_ubuntu_gnu.yml [hunk 1]
```diff
@@ -1,37 +0,0 @@
-name: Build libFMS test with autotools
```
Comment (mlee03) [inline .github/workflows/build_ubuntu_gnu.yml:1]: @mcallic2 could you merge in mixedmode one last time?

**Comment by mcallic2** [comment]
@mlee03 mixedmode merged again, its a miracle!

**Review by J-Lentz — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

**Pull request merged**